python/lsst/pipe/base/configOverrides.py

```diff
@@ -82,6 +84,21 @@ def addValueOverride(self, field, value):
         """
         self._overrides += [('value', (field, value))]
 
+    def addDatasetNameSubstitution(self, nameDictString):
+        """Add keys and values to be used in formatting config nameTemplates
+
+        This method takes in a dictionary passed in on the command line as
+        a string in the format of key:value which will be used to format
+        fields in all of the nameTemplates found in a config object. I.E.
+        a nameDictString = {'input': deep} would be used to format a
+        nameTemplate of "{input}CoaddDatasetProduct".
+        Parameters
+        ----------
+        nameDictString : str
+            String formatted as a python dict used in formatting nameTemplates
```

Comment:
I'd strongly prefer more type-safe interface here. Why no pass a `dict` here and construct that dict in `ctrl_mpexec` instead? This code is supposed to be more generic and reusable now and we should avoid depending on command-line-specific things here.